### PR TITLE
The AdminClient's verifyOrAddStore is vulnerable to connectivity issues.

### PR DESCRIPTION
**Pull request by FelixGV**
This commit makes the following changes:
- verifyOrAddStore() is now more resilient to various kinds of exceptions.
- ExceptionUtils.recursiveClassEquals() can now look for many exception types.
- VoldemortBuildAndPushJob now logs verifyOrAddStore's exceptions.

### CODE REVIEW

File: src/java/voldemort/client/protocol/admin/AdminClient.java [hunk 1]
```diff
@@ -1875,10 +1877,16 @@ public void verifyOrAddStore(StoreDefinition newStoreDef, String localProcessNam
...
-                    continue;
+                } catch (Exception e) {
+                    if (ExceptionUtils.recursiveClassEquals(e,
+                                                            UnreachableStoreException.class,
+                                                            IOException.class)) {
+                        logger.warn("Failed to contact " + node.briefToString() + " in order to validate the StoreDefinition.");
+                        unreachableNodes.add(node);
+                        continue;
+                    } else {
+                        throw new VoldemortException("Unhandled exception during verifyOrAddStore() !", e);
```
Comment (arunthirupathi) [inline src/java/voldemort/client/protocol/admin/AdminClient.java]: Can we just rethrow the e, do reduce the semantic change.

File: src/java/voldemort/client/protocol/admin/AdminClient.java [hunk 2]
```diff
@@ -1875,10 +1877,16 @@ public void verifyOrAddStore(StoreDefinition newStoreDef, String localProcessNam
                 try {
                     // Get all StoreDefinitions from each nodes in the cluster
                     remoteStoreDefs = metadataMgmtOps.getRemoteStoreDefList(nodeId).getValue();
-                } catch (UnreachableStoreException e) {
-                    logger.warn("Failed to contact " + node.briefToString() + " in order to validate the StoreDefinition.");
-                    unreachableNodes.add(node);
-                    continue;
+                } catch (Exception e) {
+                    if (ExceptionUtils.recursiveClassEquals(e,
```
Comment (arunthirupathi) [inline src/java/voldemort/client/protocol/admin/AdminClient.java]: Can you leave a comment here saying that?

Admin Client Checkout code may throw, Connect, Socket or EOFException, so checking for IOException as well.

File: src/java/voldemort/client/protocol/admin/AdminClient.java (hunk 1 above)
Comment (FelixGV) [inline src/java/voldemort/client/protocol/admin/AdminClient.java]: This function ```throws UnreachableStoreException, VoldemortException``` so it does not allow throwing the untyped ```Exception e```. Therefore, wrapping whatever ```Exception``` in ```VoldemortException``` is not a semantic change, I think.

File: src/java/voldemort/client/protocol/admin/AdminClient.java (hunk 2 above)
Comment (FelixGV) [inline src/java/voldemort/client/protocol/admin/AdminClient.java]: Done in the latest amend.

File: src/java/voldemort/client/protocol/admin/AdminClient.java (hunk 1 above)
Comment (arunthirupathi) [inline src/java/voldemort/client/protocol/admin/AdminClient.java]: Can you catch the VoldemortException then ? The Call stack shows SocketException was wrappedInside VoldemortException.

Comment (FelixGV) [inline src/java/voldemort/client/protocol/admin/AdminClient.java]: Ok, done.

**Comment by arunthirupathi** [comment]
Looks good.

**Pull request merged**